prairielib/lib/sql-db.js

```diff
@@ -203,7 +215,16 @@ module.exports.getClientAsync = async function () {
     throw new Error('Connection pool is not open');
   }
 
-  const client = await pool.connect();
+  /** @type {import('pg').PoolClient} */
+  let client;
+
+  // If we're inside a transaction, we'll reuse the same client to avoid a
+  // potential deadlock.
+  if (alsClient.getStore() === undefined) {
+    client = await pool.connect();
+  } else {
+    client = alsClient.getStore();
```

Comment:
Not that it really matters, but could we just return this `client` immediately here and avoid re-setting the search schema?

Reply:
Ignore this --- I actually read the code below and realized that we'll short-circuit because `client[SEARCH_SCHEMA]` will already equal `searchSchema`.